utils.ps1

```diff
@@ -0,0 +1,298 @@
+        }
+        if (!$importedInProfile -and !$underTest) {
+            $importedInProfile = Test-DbaToolsImportedInScript $PROFILE.CurrentUserAllHosts
+        }
+        if (!$importedInProfile -and !$underTest) {
+            $importedInProfile = Test-DbaToolsImportedInScript $PROFILE.AllUsersCurrentHost
+        }
+        if (!$importedInProfile -and !$underTest) {
+            $importedInProfile = Test-DbaToolsImportedInScript $PROFILE.AllUsersAllHosts
+        }
+
+        if ($importedInProfile) {
+            Write-Warning "Skipping add of DbaTools import to file '$profilePath'."
+            Write-Warning "DbaTools appears to already be imported in one of your profile scripts."
+            Write-Warning "If you want to force the add, use the -Force parameter."
+            return
+        }
+    }
+
+    if (!$profilePath) {
+        Write-Warning "Skipping add of DbaTools import to profile; no profile found."
+        Write-Verbose "`$profilePath          = '$profilePath'"
+        Write-Verbose "`$PROFILE              = '$PROFILE'"
+        Write-Verbose "CurrentUserCurrentHost = '$($PROFILE.CurrentUserCurrentHost)'"
+        Write-Verbose "CurrentUserAllHosts    = '$($PROFILE.CurrentUserAllHosts)'"
+        Write-Verbose "AllUsersCurrentHost    = '$($PROFILE.AllUsersCurrentHost)'"
+        Write-Verbose "AllUsersAllHosts       = '$($PROFILE.AllUsersAllHosts)'"
+        return
+    }
+
+    # If the profile script exists and is signed, then we should not modify it
+    if (Test-Path -LiteralPath $profilePath) {
+        $sig = Get-AuthenticodeSignature $profilePath
+        if ($null -ne $sig.SignerCertificate) {
+            Write-Warning "Skipping add of DbaTools import to profile; '$profilePath' appears to be signed."
+            Write-Warning "Add the command 'Import-Module DbaTools' to your profile and resign it."
+            return
+        }
+    }
+
+    # Check if the location of this module file is in the PSModulePath
+    if (Test-InPSModulePath $ModuleBasePath) {
+        $profileContent = "`nImport-Module DbaTools"
+    }
+    else {
+        $profileContent = "`nImport-Module '$ModuleBasePath\DbaTools.psd1'"
+    }
+
+    # Make sure the PowerShell profile directory exists
+    $profileDir = Split-Path $profilePath -Parent
+    if (!(Test-Path -LiteralPath $profileDir)) {
+        if ($PSCmdlet.ShouldProcess($profileDir, "Create current user PowerShell profile directory")) {
+            New-Item $profileDir -ItemType Directory -Force -Verbose:$VerbosePreference > $null
+        }
+    }
+
+    if ($PSCmdlet.ShouldProcess($profilePath, "Add 'Import-Module DbaTools' to profile")) {
+        Add-Content -LiteralPath $profilePath -Value $profileContent -Encoding UTF8
+    }
+    if ($StartSshAgent -and $PSCmdlet.ShouldProcess($profilePath, "Add 'Start-SshAgent -Quiet' to profile")) {
+        Add-Content -LiteralPath $profilePath -Value 'Start-SshAgent -Quiet' -Encoding UTF8
+    }
+}
+
+<#
+.SYNOPSIS
```

Comment:
Many of these functions will not be needed - hit me up on Slack (I'm cl) and we can work together to make install.ps1 awesome 🤗